### ISSUE TITLE
Wireframe visible when drawing a transparent object over a scalar grid

### ISSUE THREAD

**Issue by lerad** [pyvista/pyvista#5672]
### Describe the bug, what's wrong, and what you expected.

I want to visualize a scalar field on a grid. Above this grid I want to show a transparent object. 

When I draw it, there are some graphical artifacts, see below. It seems as if plotting the transparent object suddenly makes the wireframe of the scalar field grid visible. I expected that the scalar field grid in the occluded region would be as in the non-occluded region, just with a transparent object in front.

I tried several combinations of "show_edges" and "edge_opacity" but those does not seem to affect the issue.

Is this intended behavior or is this a bug? Is there some option to remove this lines?

### Steps to reproduce the bug.

```python
import numpy as np
import pyvista as pv

x = np.linspace(-10, +10, 100)
y = np.linspace(-10, +10, 100)
XX, YY, ZZ = np.meshgrid(x,y, [0])

DD = XX**2 + YY**2

grid = pv.StructuredGrid(XX, YY, ZZ)
grid.point_data['data'] = DD.flatten()

pl = pv.plotting.Plotter()
pl.add_mesh(grid)
pl.add_mesh(pv.Cube(x_length=5, y_length=5, z_length=5), color='c', opacity=0.6)
pl.screenshot('issue.png', scale=1)
pl.show()

```

### System Information

```shell
Date: Fri Feb 23 15:57:05 2024 Mitteleuropäische Zeit

                  OS : Windows
              CPU(s) : 8
             Machine : AMD64
        Architecture : 64bit
                 RAM : 13.9 GiB
         Environment : IPython
          GPU Vendor : ATI Technologies Inc.
        GPU Renderer : AMD Radeon(TM) RX Vega 10 Graphics
         GPU Version : 4.6.0 Compatibility Profile Context 22.20.16.03.220829
    MathText Support : False

  Python 3.9.13 (tags/v3.9.13:6de2ca5, May 17 2022, 16:36:42) [MSC v.1929 64
  bit (AMD64)]

             pyvista : 0.43.3
                 vtk : 9.3.0
               numpy : 1.23.4
          matplotlib : 3.6.0
              scooby : 0.9.2
               pooch : 1.8.0
              pillow : 9.3.0
             imageio : 2.22.2
           pyvistaqt : 0.11.0
               PyQt5 : 5.15.10
             IPython : 8.5.0
            colorcet : 3.0.1
             cmocean : 3.1.3
          ipywidgets : 8.0.2
               scipy : 1.9.3
                tqdm : 4.64.1
              meshio : 5.3.5
               trame : 3.5.2
        trame_client : 2.15.0
        trame_server : 2.16.1
           trame_vtk : 2.8.3
       trame_vuetify : 2.4.2
jupyter_server_proxy : 4.1.0
        nest_asyncio : 1.5.6
```

### Screenshots

![issue](https://github.com/pyvista/pyvista/assets/485326/578f4488-29b3-447d-91f8-d514b01a9a6b)
![issue_screenshot](https://github.com/pyvista/pyvista/assets/485326/37c9211c-e1e7-448f-b4d2-b80446e929a5)

**Comment by banesullivan** [comment]
unfortunately, this looks to be a windows-specific driver issue. I'm on Linux with an NVIDIA GPU without issues.

Can you try rendering these same objects in [ParaView](https://www.paraview.org/download/) to see if the issue exists there too? You can save the two meshes to read into ParaView via:

```py
grid.save('grid.vtk')
pv.Cube(x_length=5, y_length=5, z_length=5).save('cube.vtk')
```

![screenshot](https://github.com/pyvista/pyvista/assets/22067021/be4f0c89-2378-49a8-9d77-bd85bfaa30fb)

**Comment by banesullivan** [comment]
Also, does depth peeling help? https://docs.pyvista.org/version/stable/api/plotting/_autosummary/pyvista.Plotter.enable_depth_peeling.html

```py
pl.enable_depth_peeling()
```

**Comment by lerad** [comment]
Thank you for the answer! Depth peeling does not help (then the whole cube disappears). I tested the geometry in paraview and it has a slightly different but similar issue, just with different sub-components:
![snapshot-paraview](https://github.com/pyvista/pyvista/assets/485326/f1e85237-5307-473c-b91e-8f503a2ee9ff)

I also tested the same code on a different computer with Windows and an Nvidia GPU where it worked fine. I therefore concur that it is likely a driver issue. I think the issue can be closed.
